src/Runner/TestSuiteSorter.php

```diff
@@ -62,11 +106,29 @@ private function sort(TestSuite $suite, int $order, bool $resolveDependencies):
             $suite->setTests($this->randomize($suite->tests()));
         }
 
+        if ($orderDefects === self::ORDER_DEFECTS_FIRST && $this->cache !== null) {
+            $suite->setTests($this->sortDefectsFirst($suite->tests()));
+        }
+
         if ($resolveDependencies && $suite->tests()[0] instanceof TestCase) {
             $suite->setTests($this->resolveDependencies($suite->tests()));
         }
```

Comment:
from this code, it looks like it's possible to have, on single run, sorted by `randomize`, then `sortDefectsfirst`, then `resolveDependencies`. is it by purpose?